authn/keychain.go

```diff
@@ -51,14 +50,10 @@ func configDir() (string, error) {
 var errNoHomeDir = errors.New("could not determine home directory")
 
 func homeDir() string {
-	if runtime.GOOS == "windows" {
-		home := os.Getenv("HOMEDRIVE") + os.Getenv("HOMEPATH")
-		if home == "" {
-			home = os.Getenv("USERPROFILE")
-		}
-		return home
+	if h := os.Getenv("HOME"); h != "" {
+		return h
 	}
-	return os.Getenv("HOME")
+	return os.Getenv("USERPROFILE")
```

Comment:
I'd keep the explicit branch for Windows since %HOME% is also set by default on newer versions, and is often the same value as %USERPROFILE%, but not necessarily (boy oh boy was that a fun week of debugging). Specifically:
```
// dockerUserHomeDir returns the current user's home directory, as interpreted by Docker.
func dockerUserHomeDir() string {
  if runtime.GOOS == "windows" {
    // Docker specifically expands "%USERPROFILE%" on Windows, 
    return os.Getenv("USERPROFILE")	
  }
  // Docker defaults to "$HOME" Linux and OSX. 
  return os.Getenv("HOME")
}
```

Reply:
(sorry, should have just been explicit from the start)